lib/servctrl/cmd.go

```diff
@@ -153,30 +153,36 @@ func (term *ServTerm) startInteraction() {
 
 			fmt.Println(colCya + line + colRes)
 
-			// case where the server is starting
-			if ServStats.Status == "starting" {
-				// for modded server terminal compatibility, use separate check for "[Server thread/INFO]" and flag-word
-
-				// if the terminal contains flag-word "Preparing spawn area:", update ServStats.LoadProgress
-				if strings.Contains(line, "[Server thread/INFO]") && strings.Contains(line, "Preparing spawn area:") {
-					ServStats.LoadProgress = strings.Split(strings.Split(line, "Preparing spawn area: ")[1], "\n")[0]
-				}
-				// if the terminal contains flag-word "Done", the minecraft server is online
-				if strings.Contains(line, "[Server thread/INFO]") && strings.Contains(line, ": Done") {
-					ServStats.Status = "online"
-					log.Print("*** MINECRAFT SERVER IS ONLINE!")
-
-					// launch a stopInstance so that if no players connect the server will shutdown
-					RequestStopMinecraftServer()
+			// split line into a header (lineSplit[0]) and message contents (lineSplit[1]) for more robust parsing
+			var lineSplit = strings.SplitN(line, ": ", 2)
```

Comment:
can you add some code comments with an example log and why it's split in this way?

Reply:
Yep, I've written it up, let me know if anything needs further clarification. 
Also, I moved the lineSplit to be used only in the `ServStats.Status == "online"` block; I wasn't sure if I should keep it around the `"starting"` block where it isn't strictly necessary.
(and if these changes are too much, it's totally fine by me to switch to the `!contains(": <")` methods)

Reply:
ok perfect! thanksnono I think that these changes are good... anyway if in the future someone will find them too complicated it's not that big of a deal to change them